larpandoracontent/LArReclustering/ThreeDReclusteringAlgorithm.cc

```diff
@@ -0,0 +1,470 @@
+/**
+ *  @file   larpandoracontent/LArReclustering/ThreeDReclusteringAlgorithm.cc
+ *
+ *  @brief  Implementation file for the reclustering algorithm class.
+ *
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+#include "Managers/ClusterManager.h"
+
+#include "larpandoracontent/LArHelpers/LArPfoHelper.h"
+#include "larpandoracontent/LArHelpers/LArClusterHelper.h"
+#include "larpandoracontent/LArHelpers/LArPcaHelper.h"
+#include "larpandoracontent/LArHelpers/LArMCParticleHelper.h"
+
+#include "larpandoracontent/LArReclustering/ThreeDReclusteringAlgorithm.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+ThreeDReclusteringAlgorithm::ThreeDReclusteringAlgorithm():
+    m_hitThresholdForNewPfo(0),
+    m_fomThresholdForReclustering(0.3)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+ThreeDReclusteringAlgorithm::~ThreeDReclusteringAlgorithm()
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+StatusCode ThreeDReclusteringAlgorithm::Run()
+{
+    //Only proceed if successfuly get shower pfo list, and it has at least one pfo
+    const PfoList *pShowerPfoList(nullptr);
+    if ((STATUS_CODE_SUCCESS == PandoraContentApi::GetList(*this, "ShowerParticles3D", pShowerPfoList)) && pShowerPfoList->size())
+    {
+        m_newPfosListNameAllAfterReclustering = "newShowerParticles3D";
```

Comment:
Any reason this can't be in the constructor initialiser list?

Reply:
It can, and I have moved it there (and renamed it to something more legible). The only reason it was not there to begin with, was that I was initially keeping this fixed (apply reclustering to the current shower list). I suppose we can make this more flexible.